bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -0,0 +1,810 @@
+                                                                                                                 // refresh
+                                                                                                                 // every
+                                                                                                                 // x
+    // seconds
+    private final int                        cacheCount       = UPDATE_SETTINGS_INTERVAL_SECONDS
+            / UPDATE_STATUS_INTERVAL_SECONDS;                                                                    // delay
+                                                                                                                 // before
+                                                                                                                 // enabling
+                                                                                                                 // channel
+    // cache
+    private boolean                          refreshSettings  = false;
+    private boolean                          channelCache     = false;
+    protected boolean                        lockUpdates      = false;
+
+    public String                            thingName        = "";
+    private Map<String, Object>              channelData      = new HashMap<>();
+    protected ShellyBindingConfiguration     bindingConfig    = new ShellyBindingConfiguration();
+
+    /**
+     *
+     * @param thing The Thing object
+     * @param handlerFactory Handler Factory instance (will be used for event
+     *            handler registration)
+     * @param bindingConfig The binding configuration (beside thing
+     *            configuration)
+     * @param networkAddressService Network service to get local ip
+     */
+    public ShellyBaseHandler(Thing thing, ShellyHandlerFactory handlerFactory, ShellyBindingConfiguration bindingConfig,
+            @Nullable ShellyCoapServer coapServer) {
+        super(thing);
+
+        this.handlerFactory = handlerFactory;
+        this.bindingConfig = bindingConfig;
+        this.coapServer = coapServer;
+    }
+
+    /**
+     * Schedule asynchronous Thing initialization, register thing to event
+     * dispatcher
+     */
+    @Override
+    public void initialize() {
+        // start background initialization:
+        scheduler.schedule(() -> {
+            try {
+                initializeThingConfig();
+                Validate.notNull(config);
+                logger.info("{}: Device config: ipAddress={}, http user/password={}/{}, update interval={}",
+                        getThing().getLabel(), config.deviceIp, config.userId.isEmpty() ? "<non>" : config.userId,
+                        config.password.isEmpty() ? "<none>" : "***", config.updateInterval);
+
+                handlerFactory.registerDeviceListener(this);
+                initializeThing();
+            } catch (RuntimeException | IOException e) {
```

Comment:
do not catch RuntimeException

Reply:
hmm, why not? This allows me to give the user a specific information rather than a generic getStackTrace() (Maybe I'm lacking Java knowledge here)

I would like to push this out after the release with the given timeline - is that ok for you?

Reply:
As mentioned at another place: RuntimeExceptions could be everything and not at all related to your code.
So if there is a specific situation that you KNOW your binding throws a certain RuntimeException, there should be a dedicated sub-class of RuntimeException being thrown and caught here.
If you do not know of any (binding-induced) situation where a RuntimeException occurs here, simply remove the catch.

And no, please fix this before the merge - it should be pretty easy as you just have to remove the code.

Reply:
see comment on today's update below